### PR TITLE
Allow login using ssh key and generalize a little

### PR DESCRIPTION
**Pull request by miska**
Making sure that connecting to LAN/WAN device continues even if nobody asks for
password (for example because we have a key on the machine - issue #8 ) and
also generalizing a little. Although it is debian.py, 'Last login' should be
more generic then Debian motd and adding one more prompt to catch openSUSE
default root prompt which is quite minimalistic and thus there is a good chance
it will catch other prompts of various linuxes as well.

Signed-off-by: Michal Hrusecky michal.hrusecky@nic.cz

### CODE REVIEW

File: devices/debian.py [hunk 1]
```diff
@@ -55,9 +55,8 @@ def __init__(self,
             raise Exception("Unable to connect to %s." % name)
         if i == 0:
             self.sendline("yes")
-            self.expect("assword:")
-            self.sendline(password)
-        elif i == 1:
+            i = self.expect(["Last login", "assword:"])
+        if i == 1:
```
Comment (wwahammy) [inline devices/debian.py:59]: @miska any reason this was changed from elif to if?

File: devices/debian.py [hunk 2]
```diff
@@ -18,7 +18,7 @@ class DebianBox(base.BaseDevice):
     A linux machine running an ssh server.
     '''
 
-    prompt = ['root\\@.*:.*#', '/ # ', ]
+    prompt = ['root\\@.*:.*#', '/ # ', ".*:~ #" ]
```
Comment (wwahammy) [inline devices/debian.py:21]: This change should probably be in a separate commit/pull request but I don't think it should be an issue this time.

File: devices/debian.py (hunk 1 above)
Comment (miska) [inline devices/debian.py:59]: Because I deleted self.sendline(password) :-) Now the if statement on success fills i with 1 if password is needed and 0 if not and therefore the next if is a valid check whether to send a password or not.

Comment (wwahammy) [inline devices/debian.py:59]: Ah, makes sense. I see it now. Thanks for clarifying!

**Comment by wwahammy** [comment]
Unless @mbanders sees an issue, I'd support merging. Thanks @miska!
